cpp/include/cudf/column/column_device_view.cuh

```diff
@@ -472,6 +472,13 @@ class alignas(16) column_device_view : public detail::column_device_view_base {
     return d_children[child_index];
   }
 
+  /**
+   * @brief Returns the number of child columns
+   *
+   * @return The number of child columns
+   */
+  __host__ __device__ size_type num_child_columns() const noexcept { return _num_children; }
```

Comment:
Perhaps this can be `constexpr` ?

Reply:
None of the functions in that file is constexpr. I kept this as `__host__ __device__` for homogeneity. 
Perhaps a cleanup PR of `__host__ __device__` to `constexpr` might be warranted.

Reply:
https://github.com/rapidsai/cudf/issues/7713